packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -172,6 +177,96 @@ export const transformPlotlyJsonToGVBCProps = (
   };
 };
 
+// TODO: Add support for padding in continuous axis
+// FIXME: Vertical bar chart displays duplicate legend buttons
+export const transformPlotlyJsonToVBCProps = (
+  jsonObj: any,
+  colorMap: React.MutableRefObject<Map<string, string>>,
+): IVerticalBarChartProps => {
+  const { data, layout } = jsonObj;
+  const vbcData: IVerticalBarChartDataPoint[] = [];
+
+  data.forEach((series: any, index: number) => {
+    if (!series.x) {
+      return;
+    }
+
+    const scale = d3ScaleLinear()
+      .domain(d3Extent<number>(series.x) as [number, number])
+      .nice();
+    let [xMin, xMax] = scale.domain();
+
+    xMin = typeof series.xbins?.start === 'number' ? series.xbins.start : xMin;
+    xMax = typeof series.xbins?.end === 'number' ? series.xbins.end : xMax;
+
+    const bin = d3Bin().domain([xMin, xMax]);
+
+    if (typeof series.xbins?.size === 'number') {
+      const thresholds: number[] = [];
+      let th = xMin;
+      const precision = d3PrecisionFixed(series.xbins.size);
+      const format = d3Format(`.${precision}f`);
+
+      while (th < xMax + series.xbins.size) {
+        thresholds.push(parseFloat(format(th)));
+        th += series.xbins.size;
+      }
+
+      xMin = thresholds[0];
+      xMax = thresholds[thresholds.length - 1];
+      bin.domain([xMin, xMax]).thresholds(thresholds);
+    }
+
+    const buckets = bin(series.x);
+    const totalDataPoints = d3Merge(buckets).length;
```

Comment:
>d3Merge(buckets).length [](http://example.com/codeflow?start=28&length=23)

is d3Merge buckets length same as x.size

Reply:
If the start or end of [xbins](https://plotly.com/javascript/reference/histogram/#histogram-xbins) is specified, then the number of datapoints may become less than x.length

Reply:
ok. Mention this as a comment in the code